### PR TITLE
Israel.c/batch ap is exe from jnaat branch

### PR DESCRIPTION
**Pull request by IsraelCohen54**
old, irelevant

### CODE REVIEW

**Review by yaelblsch — COMMENTED** [review]
I think you should extract a data class for pet data
both in the insert nad in getPetId you pass a lot of parameter - it can be simplified if you use a data class and pass it and get the params from it

It will make also your tests more readable since the oet data can be reused instead of duplicating the same data over and over i.e. in test `addMultiplePets should add or update pets correctly`

In addition - why getPetId? query the whole pet and use whatever you need out of it

File: build.gradle.kts [hunk 1]
```diff
@@ -44,7 +44,7 @@ dependencies {
 	implementation("org.flywaydb:flyway-core")
 	implementation("org.flywaydb:flyway-database-postgresql")
 	implementation("org.jetbrains.kotlin:kotlin-reflect")
-	implementation("org.jooq:jooq:3.16.9")
+	implementation("org.jooq:jooq:3.17.9")
```
Comment (yaelblsch) [inline build.gradle.kts:47]: Don't do it now, but this kids of changes are better be done separately on their own PR

File: src/main/kotlin/com/hibob/academy/dao/PetDao.kt [hunk 2]
```diff
@@ -42,14 +41,24 @@ class PetDao(private val sql: DSLContext) {
...
-                petTable.name.eq(name)
+                petTable.id.eq(id)
                     .and(petTable.companyId.eq(companyId))
-                    .and(petTable.type.eq(type.toString()))
-                    .and(petTable.dateOfArrival.eq(dateOfArrival))
+            )
+            .fetchOne(petTableMapper)
+    }
+
+    fun getPetId(petWithoutId : PetWithoutId): Long? {
```
Comment (yaelblsch) [inline src/main/kotlin/com/hibob/academy/dao/PetDao.kt]: remove the extra space between `petWithoutId` and the `:` 

In addition - I don't understand this method - why do you fetch the id of pet by all of its attributes?
When you create the pet you fetch it all - you have the id there

File: src/main/kotlin/com/hibob/academy/filter/AuthenticationFilter.kt [hunk 3]
```diff
@@ -30,12 +30,12 @@ class AuthenticationFilter: ContainerRequestFilter {
             try {
                 Jwts.parser().setSigningKey(SECRET_KEY).parseClaimsJws(it)
             } catch (e: Exception) {
-                extracted(requestContext, e)
+                abortUnauthorized(requestContext, e)
```
Comment (yaelblsch) [inline src/main/kotlin/com/hibob/academy/filter/AuthenticationFilter.kt:33]: This change is not relevant here
It can be solved by rebase

File: src/main/kotlin/com/hibob/academy/service/PetService.kt [hunk 4]
```diff
@@ -28,4 +28,15 @@ class PetService(private val petDao: PetDao) {
...
+
+    fun adoptMultiplePets(companyId: Long, ownerId: Long, petsId: List<Long>) {
+        if (petsId.isEmpty()) {
+            throw IllegalArgumentException("Pets list cannot be empty.")
+        }
+        petDao.adoptMultiplePets(companyId, ownerId, petsId)
+    }
+
+    fun addMultiplePets(companyId: Long, petList: List<Pet>) {
+        petDao.addMultiplePets(companyId, petList)
```
Comment (yaelblsch) [inline src/main/kotlin/com/hibob/academy/service/PetService.kt]: Here the pets list can be empty?
Try to be consistent
In the former method you made sure it was not empty

File: src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt [hunk 5]
```diff
@@ -17,32 +18,41 @@ class PetDaoTest @Autowired constructor (private val sql: DSLContext) {
...
+    private val petWithoutId1 = PetWithoutId (name = "Rex", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 1), ownerId1)
+    private val petWithoutId2 = PetWithoutId (name = "Whiskers", type = PetType.CAT, companyId = companyId, dateOfArrival = LocalDate.of(2024,2,2),ownerId2)
+    private val petWithoutId3 = PetWithoutId(name = "Rexi", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 2), ownerId1)
+    private val ownerlessPetWithoutId3 = PetWithoutId(name = "Rexi", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 2), null)
+
     @Test
     fun `insert pet test`() {
 
-        petDao.insertPet(name = "Rex", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 1))
+        petDao.insertPet(petWithoutId1)
```
Comment (yaelblsch) [inline src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt]: Instead of creating and the fetching by all fields to get the id - just set the insert into a val - which will contain also the created id and use this id later

File: src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt [hunk 6]
```diff
@@ -17,32 +18,41 @@ class PetDaoTest @Autowired constructor (private val sql: DSLContext) {
...
+        val originPet = Pet(petId, petWithoutId1)
         assertEquals(originPet, insertedPet[0])
     }
 
     @Test
     fun `get pets by type test`() {
 
-        petDao.insertPet(name = "Rex", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1,1))
-        petDao.insertPet(name = "Whiskers", type = PetType.CAT, companyId = companyId, dateOfArrival = LocalDate.of(2024,2,2))
+        petDao.insertPet(petWithoutId1)
```
Comment (yaelblsch) [inline src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt]: same here 
val dog1Id = petDao.insertPet(petWithoutId1)
val cat2Id = petDao.insertPet(petWithoutId2)

and then this is redundant as you got the id from the isert
val dog1Id = petDao.getPetId(petWithoutId1)
        val cat2Id = petDao.getPetId(petWithoutId2)

File: build.gradle.kts (hunk 1 above)
Comment (IsraelCohen54) [reply build.gradle.kts:47]: got it

File: src/main/kotlin/com/hibob/academy/dao/PetDao.kt (hunk 2 above)
Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/academy/dao/PetDao.kt]: I didn't know that! amazing, would fix. Thanks!

File: src/main/kotlin/com/hibob/academy/service/PetService.kt (hunk 4 above)
Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/academy/service/PetService.kt:40]: I have been told that empty list is fine to return, hence null check is unnecessary

File: src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt (hunk 5 above)
Comment (IsraelCohen54) [reply src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt]: Amazing, would fix on next push.

File: src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt (hunk 6 above)
Comment (IsraelCohen54) [reply src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt]: fixed, amazing feedback

File: src/main/kotlin/com/hibob/academy/service/PetService.kt (hunk 4 above)
Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/academy/service/PetService.kt]: fixed

File: src/main/kotlin/com/hibob/academy/resource/PetController.kt
Comment (yaelblsch) [inline src/main/kotlin/com/hibob/academy/resource/PetController.kt:1]: Please rename PetController.kt
In kotlin projects we call those With Resource suffix => PetResource.kt

File: src/main/kotlin/com/hibob/academy/resource/PetController.kt [hunk 7]
```diff
@@ -69,19 +73,37 @@ class PetController (private val petService: PetService) {
...
+    fun adoptMultiplePets(
+        @PathParam("ownerId") ownerId: Long,
+        @PathParam("companyId") companyId: Long,
+        petList: List<Long>
+    ): Response {
+        petService.adoptMultiplePets(ownerId, companyId, petList)
+        return Response.ok().build()
+    }
+
+    @PUT
```
Comment (yaelblsch) [inline src/main/kotlin/com/hibob/academy/resource/PetController.kt]: This is creation and not edit method so it should be `POST` instead of `PUT`

File: src/main/kotlin/com/hibob/academy/dao/PetDao.kt [hunk 8]
```diff
@@ -24,14 +23,15 @@ class PetDao(private val sql: DSLContext) {
...
 
-    fun insertPet(name: String, type: PetType, companyId: Long, dateOfArrival: LocalDate, ownerId: Long? = null) {
-        sql.insertInto(petTable)
-            .set(petTable.name, name)
-            .set(petTable.type, type.toString())
-            .set(petTable.companyId, companyId)
-            .set(petTable.dateOfArrival, dateOfArrival)
-            .set(petTable.ownerId, ownerId)
-            .execute()
+    fun insertPet(petWithoutId: PetWithoutId): Long? {
```
Comment (yaelblsch) [inline src/main/kotlin/com/hibob/academy/dao/PetDao.kt]: We prefer not to return nullable id - if it is null - then we have and issue and maybe it would be better to throw an exception this case
If it is not null - just return the is as Long and not Long?

File: src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt [hunk 9]
```diff
@@ -95,11 +145,16 @@ class PetDaoTest @Autowired constructor (private val sql: DSLContext) {
         ownerDao.insertOwner("Bob1", companyId, "123")
 
         val ownerId = ownerDao.getOwnerId(companyId, "123")
```
Comment (yaelblsch) [inline src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt]: Not sure I understand this 
You add an owner called Bob1 and get it id  as owner id
What is this `val ownerId = ownerDao.getOwnerId(companyId, "123")`
Is it the same as the id we now return on pet? maybe it would be better to change that insertOwner will return its id

File: src/main/kotlin/com/hibob/academy/filter/AuthenticationFilter.kt (hunk 3 above)
Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/academy/filter/AuthenticationFilter.kt:33]: Changed it back and opened a new branch and did it there (waiting for Eli).
Fixed

File: src/main/kotlin/com/hibob/academy/resource/PetController.kt
Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/academy/resource/PetController.kt:1]: thanks

File: src/main/kotlin/com/hibob/academy/resource/PetController.kt (hunk 7 above)
Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/academy/resource/PetController.kt]: fixed

File: src/main/kotlin/com/hibob/academy/dao/PetDao.kt (hunk 8 above)
Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/academy/dao/PetDao.kt]: got it, fixed

File: src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt (hunk 9 above)
Comment (IsraelCohen54) [reply src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt]: fixed (it returns owner id (- "ownerDao.getOwnerId"), missed to fix it after I ficed pedDao, I use it to compare owner insertion)

File: src/main/kotlin/com/hibob/academy/resource/PetResource.kt [hunk 10]
```diff
@@ -60,7 +60,11 @@ class PetController (private val petService: PetService) {
 
     @POST
```
Comment (yaelblsch) [inline src/main/kotlin/com/hibob/academy/resource/PetResource.kt]: This is update - not create - so should be @PUT and not @POST

**Review by yaelblsch — APPROVED** [review]
I've approved - but pay attention to my single comment- regarding code not related to you new changes - please fix

File: src/main/kotlin/com/hibob/academy/resource/PetResource.kt (hunk 10 above)
Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/academy/resource/PetResource.kt]: fixed
